src/test/java/minijava/CliTest.java

```diff
@@ -14,20 +17,33 @@
 
 	/**
 	 * Tests the `--echo` option of the cli by calling the `run` script
-     */
+	 */
 	@Test
 	public void echo() throws Exception {
 		Process process = Runtime.getRuntime().exec(new String[]{"./run", "--echo", "non_existing_file"});
 		process.waitFor();
 		assertEquals("Error code for invalid --echo call", 1, process.exitValue());
+
 		File tmpFile = File.createTempFile("tmp", "tmp");
+		tmpFile.deleteOnExit();
 		FileWriter writer = new FileWriter(tmpFile);
 		writer.write("t");
 		writer.close();
 		process = Runtime.getRuntime().exec(new String[]{"./run", "--echo", tmpFile.getAbsolutePath()});
 		process.waitFor();
 		assertEquals(0, process.exitValue());
 		assertEquals('t', process.getInputStream().read());
-		tmpFile.delete();
+
+		File tmpDirectory = Files.createTempDirectory("test").toFile();
```

Comment:
It's preferable to have multiple small tests instead of one big one. The advantage is that you can immediately see in the test report what's wrong (if the test cases have a expressive names, say "runEchoWithProjectDirAsWorkingDirectory" and "runEchoOptionWithSomeTempDirAsWorkingDirectory")